### ISSUE TITLE
Resume training - first iteration frequently produces a "best" test loss.

### ISSUE THREAD

**Issue by lukeforehand** [lukeforehand/deertracker#29]
I am seeing that upon resuming a classifier training, the first iteration at epoch X seems to produce a better test loss than the past hundred or more epochs.  It happens enough to make me suspicious.  Wondering if @kbrose has any thoughts on this?

**Comment by lukeforehand** [comment]
I'm wondering if with each resume, the validation_split is changing which files are part of the test set, so the model is accidentally getting tested on data that it was trained on.

**Comment by kbrose** [comment]
I think your intuition is right, the split is probably changing between restarts. I don't think the cause would be the `shuffle` parameter, and I did [fix the seed manually](https://github.com/lukeforehand/deertracker/blob/main/deertracker/classifier.py#L94), but if you are adding more data in between runs that would probably cause the train/test split to be different even with the same fixed seed (in the same way `random.Random(1).choice([0, 1, 2]) != random.Random(1).choice([0, 1, 2, 3])`).

The stablest way to do this may be to do our own train/test split, e.g. have a structure like

```
train/
  bear/
  deer/
  ...
test/
  bear/
  deer/
  ...
```

That way when you restart you can be sure you're keeping the same split. The `get_datasets` function would need updated:

```python
    train_ds = tf.keras.preprocessing.image_dataset_from_directory(
        data_dir / 'train',
        validation_split=0,  # did our own split, so set keras's split to 0
        subset="training",
        seed=20201130,
        image_size=(IMAGE_SIZE + 10, IMAGE_SIZE + 10),
        batch_size=1,
    )
    test_ds = tf.keras.preprocessing.image_dataset_from_directory(
        data_dir / 'test',
        validation_split=0,
        subset="training",  # pulling from the test folder, so pretend it's "training" subset
        seed=20201130,
        image_size=(IMAGE_SIZE, IMAGE_SIZE),
        batch_size=1,
    )
```

**Comment by lukeforehand** [comment]
Excellent thank you @kbrose!  I'll make this change to the function that exports the training dataset and see what happens.

I was thinking of a sophisticated way to consistently select the train vs test set from the database, even in between additions of new data, but seems a little complicated.  So maybe best that I don't add new data in between resumes of training.
